__test__/utils/scraper.test.ts

```diff
@@ -0,0 +1,173 @@
+import type { SerpApiSearchResults, SearchResult } from "../../types.d";
+import { ScraperType } from "../../types.d";
+
+import { scrapeKeywordFromGoogle, UNKNOWN_ERROR } from "../../utils/scraper";
+
+import { getDummyKeywords } from "../data";
+
+global.fetch = jest.fn();
+
+beforeEach(() => {
+  fetch.mockClear();
+});
+
+const getDummySettings = () => ({
+  scraper_type: ScraperType.SERPAPI,
+  scaping_api: "42",
+  notification_interval: "never",
+  notification_email: "",
+  notification_email_from: "",
+  smtp_server: "",
+  smtp_port: "",
+  smtp_username: "",
+  smtp_password: "",
+});
+
+describe("#scrapeKeywordFromGoogle", () => {
+  it("Makes a request to SerpApi", async () => {
+    const keyword = getDummyKeywords()[0];
+
+    await scrapeKeywordFromGoogle(keyword, getDummySettings());
+
+    expect(global.fetch).toHaveBeenCalledTimes(1);
+    expect(global.fetch).toHaveBeenCalledWith(
+      expect.stringMatching("https://serpapi.com/search"),
+      expect.objectContaining({
+        headers: {
+          "Content-Type": "application/json",
+          "User-Agent": expect.stringMatching("Mozilla/5.0"),
+          Accept: "application/json; charset=utf8;",
+        },
+      })
+    );
+  });
+
+  describe("When SerpApi returns a successful response", () => {
+    it("Extracts refreshed SERP result", async () => {
+      const position = Math.ceil(Math.random() * 100);
+
+      const matchedOrganicResult = {
+        title: `image compressor result on position ${position}`,
+        link: `https://compressimage.io/matched/result-${position}`,
+        position,
+
+        snippet: "TEST",
+        displayed_link: "TEST",
+      };
+
+      const dummySerpApiResponse: SerpApiSearchResults = {
+        search_metadata: { id: "TEST", status: "Success" },
+        organic_results: [matchedOrganicResult],
+      };
+
+      fetch.mockImplementation(() =>
+        Promise.resolve({
+          json: () => Promise.resolve(dummySerpApiResponse),
+        })
+      );
+
+      const matchedResult = {
+        title: matchedOrganicResult.title,
+        url: matchedOrganicResult.link,
+        position: matchedOrganicResult.position,
+      };
+
+      const lastResult: SearchResult[] = [matchedResult];
```

Comment:
Test setup contains 30 lines of code because `scrapeKeywordFromGoogle` uses `extractScrapedResult` and `getSerp`.

This integration test to makes sure that `scrapeKeywordFromGoogle` as a whole, without any method stubs, except `fetch` in `getScraperClient`.